metron-platform/metron-common/src/main/java/org/apache/metron/common/query/PredicateProcessor.java

```diff
@@ -18,12 +18,20 @@
 
 package org.apache.metron.common.query;
 
-import org.antlr.v4.runtime.*;
-import org.apache.metron.common.query.generated.*;
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.TokenStream;
+import org.apache.metron.common.query.generated.PredicateLexer;
+import org.apache.metron.common.query.generated.PredicateParser;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
 
 
 public class PredicateProcessor {
   public boolean parse(String rule, VariableResolver resolver) {
+    if (isEmpty(rule)) {
```

Comment:
Right, I wasn't worried about empty values (your case covered that); I was worried about whitespace-only queries.  In that case `PredicateProcessor.parse("  ") != PredicateParser.parse("")`